ngrinder-controller/src/main/java/org/ngrinder/infra/webhook/service/WebhookService.java

```diff
@@ -0,0 +1,119 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ngrinder.infra.webhook.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.ngrinder.infra.webhook.model.Event;
+import org.ngrinder.infra.webhook.model.WebhookActivation;
+import org.ngrinder.infra.webhook.model.WebhookConfig;
+import org.ngrinder.model.PerfTest;
+import org.ngrinder.model.User;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static java.time.Duration.ofSeconds;
+import static org.ngrinder.common.util.AccessUtils.getSafe;
+import static org.ngrinder.common.util.CollectionUtils.newHashMap;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class WebhookService {
+
+	private final NgrinderWebhookClient ngrinderWebhookClient;
+
+	private final WebhookActivationService webhookActivationService;
+
+	private final ObjectMapper objectMapper;
+
+	public void sendWebhookRequest(PerfTest perfTest,
+								   WebhookConfig webhookConfig,
+								   Event event) {
+
+		Map<String, Object> payLoad = event.getPayloadBuilder().apply(perfTest);
+		Consumer<HttpHeaders> headers = httpHeaders -> httpHeaders.add("X-nGrinder-Event", event.name());
+		ResponseEntity<String> responseEntity = null;
+
+		try {
+			Mono<ResponseEntity<String>> responseEntityMono
+				= ngrinderWebhookClient.post(webhookConfig.getPayloadUrl(), webhookConfig.getContentType().getMediaType(), payLoad, headers);
+			responseEntity = responseEntityMono.block(ofSeconds(2));
+		} catch (RuntimeException e) {
+			responseEntity = new ResponseEntity<>("An exception occurred while sending the webhook request.\n" + e.getMessage(), BAD_REQUEST);
+		} finally {
+			saveWebhookActivation(perfTest, webhookConfig, event, payLoad, responseEntity);
+		}
+	}
+
+	public void sendDummyWebhookRequest(User user, WebhookConfig webhookConfig, Event event) {
+		PerfTest perfTest = new PerfTest();
+		perfTest.setCreatedUser(user);
+		sendWebhookRequest(perfTest, webhookConfig, event);
+	}
+
+	private void saveWebhookActivation(PerfTest perfTest,
+									   WebhookConfig webhookConfig,
+									   Event event,
+									   Map<String, Object> request,
+									   ResponseEntity<String> responseEntity) {
+
+		Map<String, Object> header = newHashMap();
+		header.put("X-nGrinder-Event", event.name());
+		header.put("Content-Type", webhookConfig.getContentType().getName());

```

Comment:
We may need to make these to constants

```java
String NGRINDER_WEBHOOK_EVENT_HEADER = "X-nGrinder-Event"
```

but, example above is quite longer than original value. I'm not sure what is the best constant naming.

Reply:
fixed 

https://github.com/naver/ngrinder/blob/447a3f6a99ed5c8a5f5634deca0be9662b5f5f1f/ngrinder-controller/src/main/java/org/ngrinder/infra/webhook/service/WebhookService.java#L92

https://github.com/naver/ngrinder/blob/447a3f6a99ed5c8a5f5634deca0be9662b5f5f1f/ngrinder-controller/src/main/java/org/ngrinder/infra/webhook/service/WebhookService.java#L44

https://github.com/naver/ngrinder/blob/447a3f6a99ed5c8a5f5634deca0be9662b5f5f1f/ngrinder-controller/src/main/java/org/ngrinder/infra/webhook/service/WebhookService.java#L93